homeassistant/components/google_travel_time/__init__.py

```diff
@@ -15,3 +21,25 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Unload a config entry."""
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry):
```

Comment:
We should make this a soft-migration and do this inside the async_setup_entry method

Reply:
You can check if you need migration by cheking if the config entry has a unique ID.

Reply:
I think that the issue you experienced in development is because of a race condition. I would suggest you create a migration/import lock.

```python
if entry.unique_id:
    async with hass.data.setdefault(DOMAIN, asyncio.Lock()):
        # Do migration
        ...
```

You should do the same inside the config flow import step.